shell/platform/windows/windows_lifecycle_manager.h

```diff
@@ -0,0 +1,33 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_WINDOWS_WINDOW_TOP_LEVEL_MESSAGE_HANDLER_H_
+#define FLUTTER_SHELL_PLATFORM_WINDOWS_WINDOW_TOP_LEVEL_MESSAGE_HANDLER_H_
+
+#include <cstdint>
+
+#include <Windows.h>
+
+namespace flutter {
+
+class FlutterWindowsEngine;
+
+class WindowsLifecycleManager {
+ public:
+  WindowsLifecycleManager(FlutterWindowsEngine& engine);
```

Comment:
Since we've got a reference cycle between `FlutterWindowsEngine` and `WindowsLifecycleManager` this should be a weak pointer (just a raw `FlutterWindowsEngine*` pointer) to indicate that to the reader.